### PR TITLE
Test

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
cc @gkoz

### CODE REVIEW

File: tests/tests.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+              b: u32) {}
+<!-- mod Bar§macro test!§struct SuperFoo -->
+ struct inside macro
+<!-- mod Bar§macro test!§macro sub_test!§struct FooFoo -->
+ and another one!
+"#, file, "{}")
+}
+
+fn gen_file(filename: &str, content: &str) -> File {
+    match OpenOptions::new().write(true).create(true).truncate(true).open(&format!("{}/{}", TEST_FILE_DIR, filename)) {
```
Comment (gkoz) [inline tests/tests.rs]: `File::create`?

File: tests/tests.rs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        Ok(mut f) => {
+            write!(f, "{}", content).unwrap();
+            f
+        },
+        Err(e) => {
+            panic!("gen_file: {}", e)
```
Comment (gkoz) [inline tests/tests.rs]: See `Result::expect`.

File: tests/tests.rs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    let comment_file = "basic.cmts";
+    create_dir(TEST_FILE_DIR);
+    {
+        gen_file(test_file, BASIC);
+        let mut f = gen_file(comment_file, "");
+        stripper_lib::strip_comments(Path::new(TEST_FILE_DIR), test_file, &mut f, false);
+    }
+    compare_files(&get_basic_cmt(test_file), &format!("{}/{}", TEST_FILE_DIR, comment_file));
+    compare_files(BASIC_STRIPPED, &format!("{}/{}", TEST_FILE_DIR, test_file));
+    clean_test(&vec!(test_file, comment_file));
```
Comment (gkoz) [inline tests/tests.rs]: It's usually a good idea to clean before the test not after.

Unnecessary vector, huh, that's some strong habit.

File: tests/tests.rs [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    A: u32,
+}
+
+mod Bar {
+    test! {
+        /// struct inside macro
+        struct SuperFoo;
+        sub_test! {
+            /// and another one!
+            struct FooFoo {
```
Comment (gkoz) [inline tests/tests.rs]: Doesn't look like the kind of scenario we might expect.

**Comment by gkoz** [comment]
Wow, [look here](https://travis-ci.org/GuillaumeGomez/rustdoc-stripper/jobs/110634697)
We're going to get Ubuntu 14.04 at last! *not two months before 16.04 comes out...*

**Comment by gkoz** [comment]
You should remove `sudo: true` from `.travis.yml` by the way.

File: tests/tests.rs (hunk 4 above)
Comment (GuillaumeGomez) [inline tests/tests.rs]: ?

**Comment by GuillaumeGomez** [comment]
> Wow, look here
> We're going to get Ubuntu 14.04 at last! not two months before 16.04 comes out...

Better later than never haha.

File: tests/tests.rs (hunk 3 above)
Comment (GuillaumeGomez) [inline tests/tests.rs]: I prefer using slices instead of `Vec` when I just need read access.

Comment (gkoz) [inline tests/tests.rs]: `vec!()` is not exactly a slice.

Comment (GuillaumeGomez) [inline tests/tests.rs]: `vec!()` is `Vec` which I transform into `&[]`.

File: tests/tests.rs (hunk 4 above)
Comment (gkoz) [inline tests/tests.rs]: It's not realistic to have this kind of nesting. This and the `mod Bar§macro test!§struct SuperFoo` feature they're lines of code that are going to sit there unused, getting in the way of features we need and bug fixes.

File: tests/tests.rs (hunk 3 above)
Comment (gkoz) [inline tests/tests.rs]: Just take a `&[]`?

Comment (GuillaumeGomez) [inline tests/tests.rs]: I feel stupid...

File: tests/tests.rs (hunk 4 above)
Comment (GuillaumeGomez) [inline tests/tests.rs]: I think it's better to keep it this way for the moment and then we'll update it when needed. For now we're trying to maintain the current implementation.

**Pull request merged**